data-tx-hibernate/src/main/java/io/micronaut/transaction/hibernate/HibernateTransactionManager.java

```diff
@@ -203,6 +206,51 @@ protected void doRollback(DefaultTransactionStatus<Session> tx) {
         }
     }
 
+    @Override
+    protected void doNestedBegin(DefaultTransactionStatus<Session> status) {
+        try {
+            Session session = status.getConnection();
+            Savepoint savepoint = getConnection(session).setSavepoint(status.getTransactionDefinition().getName());
```

Comment:
doesn't this basically skip over hibernate and just does nested tx at the jdbc level? is this safe wrt hibernate internal session caches?

Reply:
Yeah, that's how Spring does it